keps/sig-scheduling/20190926-default-even-pod-spreading.md

```diff
@@ -113,87 +116,82 @@ As a workload author, I want to spread the workload in the cluster, but:
+`DefaultPodTopologySpread`. Thus, there's no need for the features to co-exist.
 
-| SelectorSpreading | EvenPodsSpread | Operator constraints | Valid | Pod spread constraints        |
-| :---------------: | :------------: | :------------------: | :---: | :---------------------------: |
-| N                 | Y              | Not provided         | Yes   | [k8s default](#default-rules) |
-| N                 | Y              | Provided             | Yes   | provided by operator          |
-| Y                 | Y              | Not provided         | Yes   | [k8s default](#default-rules) |
-| Y                 | Y              | Provided             | No    | -                             |
-| N                 | N              | -                    | Yes   | None                          |
-| Y                 | N              | -                    | No    | -                             |
+When the feature gate is enabled:
 
-Selecting `SelectorSpreadingPriority` but not `EvenPodsSpreadPriority` in a policy is an invalid
-configuration, because the latter is a requirement for the former.
+- K8s will set Default `topologySpreadConstraints` and remove `DefaultPodTopologySpread` from the
+k8s providers (`DefaultProvider` and `ClusterAutoscalerProvider`). The
+[Default](#default-constraints) will have a similar effect.
+- When a policy is used, `SelectorSpreadingPriority` will map to `PodTopologySpread`.
+- When setting plugins in the Component Config API, plugins are added as requested. Since an
+  operator is manually enabling the plugins, we assume they are aware of their intentions.
 
 ### Risks and Mitigations
 
-`EvenPodsSpreadPriority` has some overhead and we currently ensure that pods that don't use the
-feature get minimally affected. After Default `topologySpreadConstraints` is rolled out,
-all pods will run through the algorithms.
-However, we should ensure that the running overhead is not significantly higher than
-`SelectorSpreadingPriority` with k8s Default `topologySpreadConstraints`.
+The `PodTopologySpread` plugin has some overhead compared to other plugins. We currently ensure that
+pods that don't use the feature get minimally affected. After Default `topologySpreadConstraints`
+is rolled out, all pods will run through the plugin.
+We should ensure that the running overhead is not significantly higher than
+`DefaultPodTopologySpread` with the k8s Default.
 
 ## Design Details
 
 ### API
 
-A new structure called `TopologySpreadConstraint` is introduced to `KubeSchedulerConfiguration`.
+A new structure `Args` is introduced in `pkg/scheduler/framework/plugins/podtopologyspread`.
+Values are decoded from the `pluginConfig` slice in the kube-scheduler Component Config and used in
+`podtopologyspread.New`.
 
 ```go
-type KubeSchedulerConfiguration struct {
-	....
-	// TopologySpreadConstraints defines topology spread constraints to be applied to pods
+type Args struct {
```

Comment:
How about adding a file name as a comment. Because the definition was in CC API before.

```go
// pkg/scheduler/framework/plugins/podtopologyspread/filtering.go
```

Reply:
Done